contracts/VotiumBribesProcessor.sol

```diff
@@ -129,10 +130,22 @@ contract VotiumBribesProcessor is CowSwapSeller {
 
     /// === Helper Functions === ///
 
-    function _lastSchedule(address _token) internal view returns (uint256 endingTime) {
+    function _getScheduleTimes(address _token) internal view returns (
+        uint256 newStartTime, uint256 newEndTime, uint256 duration
+    ) {
         IRewardsLogger.UnlockSchedule[] memory schedules = REWARDS_LOGGER
             .getUnlockSchedulesFor(address(BVE_CVX), _token);
-        endingTime = schedules[schedules.length - 1].end;
+        uint256 lastEndingTime = schedules[schedules.length - 1].end;
+        uint256 minStartTime = BADGER_TREE.lastPublishTimestamp();
+        if(minStartTime <= lastEndingTime) {
```

Comment:
i envisioned will be the condition on the opposite way, say:

- if we run `swapCVXTobveCVXAndEmit` & `emitBadger` way before the **lastEndingTime** (which may be the case if bribes are process promptly), then the starting time should be **lastEndingTime**, otherwise running before the last schedule ended two schedules will overlap. Since:

```
newStartTime = minStartTime + 4 hours;
```

may be way before than `lastEndingTime`.

Then, if when running these two methods and `minStartTime` is **greater** than `lastEndingTime` means that schedules were delayed for some reason and we should post the schedules with the `starting time = minStartTime + GRACE_CYCLE_TIME` (being the grace cycle time 4h).

Another case, perhaps to check is when `minStartTime + 2h` may fall within the `lastEndingTime`, since then some portions of the rewards there will not be distributed properly if we use `lastEndingTime` as `newStartTime`.

Perhaps, another checkup worth adding is for example in the `minStartTime <= lastEndingTime`,